File: 프로그래머스/연습문제/Lv2/N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> arr) {
    // 내림차순 정렬
    sort(arr.begin(), arr.end(), greater<int>());
    
    // 가장 큰 수의 배수 중에서 최소공배수 찾기
    for(int i=arr[0]; ; i+=arr[0]){
        bool flag = true;
        for(int j=1; j<arr.size(); j++){
            if(i%arr[j]){
                flag = false;
                break;
            }
        }
        
        if(flag) return i;
    }
}

File: 프로그래머스/연습문제/Lv2/N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    # 내림차순 정렬
    arr = sorted(arr, reverse=True)

    # 가장 큰 수의 배수 중에서 최소공배수 찾기
    i = arr[0]
    while True:
        flag = True
        for j in range(1, len(arr)):
            if i%arr[j]:
                flag = False
                break

        if flag: return i
        else: i += arr[0]
